File: sort/quicksort.cpp
#include<iostream>
using namespace std;
void quickSort(int a[],int,int);

int main(){
    int array[]={34,65,12,43,67,5,78,10,3,70};
    int k;
    int len=sizeof(array)/sizeof(int);
    quickSort(array,0,len-1);

}

void quickSort(int array[],int left,int right){
    if (left<right)
    {
    int i=left;
    int j=right;
    int key=array[i];
    while(i<j){
        while(i<j && array[j]>=key){
            j--;
        }
        if(i<j){
            array[i]=array[j];
        }
        while(i<j&& array[i]<=key)
        {
            i++;
        }
        if (i<j){
            array[j]=array[i];
        }
    }
    array[i]=key;
    quickSort(array,left,i-1);
    quickSort(array,i+1,right);
    }

}

File: sort/quicksort.py

def quicksort(arr):
    if len(arr)<=1:
        return arr
    pivot=arr[-1]
    left=[]
    right=[]
    for element in arr[:-1]:
        if element<=pivot:
            left.append(element)
        else:
            right.append(element)
            
    return quicksort(left)+[pivot]+quicksort(right)
            

arr=[10,7,8,9,1,5]

sorted_arr=quicksort(arr)
print(sorted_arr)
